### PR TITLE
201 extend header with lobby display copy button

### PR DESCRIPTION
**Pull request by 0815Sailsman**
Still WIP

### CODE REVIEW

**Review by Straczi — CHANGES_REQUESTED** [review]
as user i would like to have some kind of feedback when the lobbyId has been copied into my clipboard. Maybe add some onHover text saying "copy lobbyID to clipboard" and display a small toast or a small message above the button saying "copied!"

Edit: also a little bit more feedback that the button is actually a button and clickable would be nice (maybe in form of a small onHover effect that makes the button a litte bit more grey)

File: frontend/app/src/components/header/Header.tsx [hunk 1]
```diff
@@ -2,10 +2,32 @@ import "./Header.css";
...
+	const lobby: Lobby | null = useLobby();
+
+	const lobbyDispatcher = useDispatchLobby();
+	useEffect(() => {
+		if (lobbyDispatcher) {
+			lobbyDispatcher({
+				type: "join",
+				lobby: new Lobby(0, [], GameState.NOT_STARTED),
+			});
+		}
```
[the remark this answers is not in the document]
Comment (0815Sailsman) [reply frontend/app/src/components/header/Header.tsx:25]: yep, this was for testing purposes and I forgot to remove it

File: frontend/app/src/TestFactories.ts [hunk 2]
```diff
@@ -1,8 +1,18 @@
 import { Lobby } from "@/domain/Lobby.ts";
 import { GameState } from "@/domain/GameState.ts";
 
-export function aLobby(): Lobby {
-	return new Lobby(0, ["playerA"], GameState.NOT_STARTED);
+interface NamedLobbyParameters {
+	lobbyId?: number;
+	playersJoined?: string[];
+	gameState?: GameState;
+}
```
[the remark this answers is not in the document]
Comment (0815Sailsman) [reply frontend/app/src/TestFactories.ts:8]: No particular reason, could also inline the types 🤷🏻

[the remark this answers is not in the document]
Comment (0815Sailsman) [reply frontend/app/src/TestFactories.ts:8]: but I think that looks a little too cluttered tbh

**Pull request merged**